Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
 
 ### Creating Custom MRML Node Classes
 
-- Custom MRML nodes provide persistent storage for the module parameters.
-- Custom MRML nodes should be registered with the MRML scene using `RegisterNodeClass()` so they can be saved and restored from a scene file.
-- Classes should implement the following methods:
-  - `CreateNodeInstance()` similar to VTK New() method only not static.
-  - `GetNodeTagName()` returns a unique XML tag for this node.
-  - `ReadXMLAttributes()` reads node attributes from XML file as name-value pairs.
-  - `WriteXML()` writes node attributes to output stream.
-  - `Copy()` – copies node attributes.
+If you are adding new functionality to 3D Slicer either via extensions, or even updates to the core, most of the time the existing MRML nodes will be sufficient. Many powerful C++ and Python extensions simply use and combine existing the node types to create new functionality. Other extensions derive from existing classes and add just a few methods to get the functionality they need, instead of creating new MRML types from scratch. If the existing MRML nodes don't offer enough (or almost enough) functionality to enable what you want to do, it is possible to create your own MRML nodes with a little bit of effort.
+
+There are a number of different MRML nodes and helper classes that can be implemented to enable new MRML data type functionality. Here is the not-so-short list. We will go over each of these in detail. 
+
+1. [Data node](#the-data-node)
+2. [Display node](#the-display-node)
+3. [Widget](#the-widget)
+4. [Widget Representation](#the-widget-representation)
+5. [Displayable Manager](#the-displayable-manager)
+6. [Storage node](#the-storage-node)
+7. [Reader](#the-reader)
+8. [Writer](#the-writer)
+9. [Subject Hierarchy Plugin](#the-subject-hierarchy-plugin)
+10. [Module](#the-module-aka-putting-it-all-together)
+
+While technically not all of these need to be implemented for a new MRML type to be used and useful, implementing all of them will yield the best results. The resulting MRML type will “be like the Model”.
+
+For the filenames and classes, replace \<MyCustomType> with the name of your type.
+
+:::{note}
+
+MRML nodes are implemented in C++.
+
+
+MRML nodes can be implemented in a Slicer extension.
+:::
+
+:::{note}
+
+All links to classes/functions are to the current head of Slicer.
+
+:::
+
+#### The data node
+
+The data node is where the essence of the new MRML type will live. It is where the actual data resides. Notably absent from the data node is any description of how the data should be displayed or stored on disk (with the exception of MRB files).
+
+Files:
+
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>Node.h
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>Node.cxx
```

Comment:
Was initially thinking about using 

```
* `<Extension>/<Module>/MRML/vtkMRML<MyCustomType>Node.h`
* `<Extension>/<Module>/MRML/vtkMRML<MyCustomType>Node.cxx`
```

but formatting it like this may be best:

```
|-- <Extension>
       |-- <Module>
              |-- MRML
                    |-- vtkMRML<MyCustomType>Node.h
                    |-- vtkMRML<MyCustomType>Node.cxx
```

Reply:
Updated to suggested file structure formatting.